modules/nextflow/src/test/groovy/nextflow/k8s/K8sDriverLauncherTest.groovy

```diff
@@ -192,6 +192,73 @@ class K8sDriverLauncherTest extends Specification {
 
     }
 
+    def 'should create launcher spec job' () {
+
+        given:
+        def pod = Mock(PodOptions)
+        pod.getVolumeClaims() >> [ new PodVolumeClaim('pvc-1', '/mnt/path/data') ]
+        pod.getMountConfigMaps() >> [ new PodMountConfig('cfg-2', '/mnt/path/cfg') ]
+        pod.getAutomountServiceAccountToken() >> true
+
+        def k8s = Mock(K8sConfig)
+        k8s.getNextflowImageName() >> 'the-image'
+        k8s.getLaunchDir() >> '/the/user/dir'
+        k8s.getWorkDir() >> '/the/work/dir'
+        k8s.getProjectDir() >> '/the/project/dir'
+        k8s.getPodOptions() >> pod
+        k8s.getJob() >> true
+
+        and:
+        def driver = Spy(K8sDriverLauncher)
+        driver.@runName = 'foo-boo'
+        driver.@k8sClient = new K8sClient(new ClientConfig(namespace: 'foo', serviceAccount: 'bar'))
+        driver.@k8sConfig = k8s
+
+        when:
+        def spec = driver.makeLauncherSpec()
+        then:
+        driver.getLaunchCli() >> 'nextflow run foo'
+
+        spec == [
+            apiVersion: 'batch/v1', 
+            kind: 'Job', 
+            metadata: [name: 'foo-boo', namespace: 'foo'], 
+            spec: [
+               backoffLimit: 0,
+               template: [
+                  apiVersion: 'v1',
+                  kind: 'Pod',
+                  metadata: [labels:[app:'nextflow', runName:'foo-boo']],
+                  spec: [
```

Comment:
Looking at the `Job` [spec](https://kubernetes.io/docs/concepts/workloads/controllers/job/) it looks those attributes should not be added. 

I'm going to submit a patch to improve it